### DISCUSSION TITLE
Using avalanche in older python and pytorch version

### DISCUSSION THREAD

**Discussion by SinHanYang** [ContinualAI/avalanche#1110] [Q&A] [answered]
Hi, I would like to combine your product with another project X. However, project X works in python 3.6 and pytorch 1.6.0.
When I tried to use avalanche in this environment, I shows error:
```
File "replay.py", line 31, in <module> from avalanche.benchmarks import nc_scenario
...
class SubsetWithTargets(Generic[T_co, TTargetType], Subset[T_co]): TypeError: 'type' object is not subscriptable
```
Just like discussion #515.

I followed the comments in the discussion:
on line 63 in avalanche_dataset.py:
`class AvalancheDataset(IDatasetWithTargets[T_co, TTargetType], Dataset[T_co]):`
changes to
`class AvalancheDataset(IDatasetWithTargets[T_co, TTargetType], Dataset):`

and on line 173 in dataset_utils.py:
`class SubsetWithTargets(Generic[T_co, TTargetType], Subset[T_co]):`
changes to
`class SubsetWithTargets(Generic[T_co, TTargetType], Subset):`

However, the code showed error `can't find T_co`.

Therefore, it's possible to run avalanche in older python and pytorch version without changing anything?

Thanks

**Comment by vlomonaco** [comment, marked as answer]
Hi @SinHanYang, sorry for the late reply.
We have just dropped the support for py3.6. 

You may want to use an older Avalanche version or update your code!
